src/components/DevicesModaleConfigureView.jsx

```diff
@@ -0,0 +1,338 @@
+import IconButton from 'cozy-ui/transpiled/react/IconButton'
+import List from 'cozy-ui/transpiled/react/MuiCozyTheme/List'
+import ListItem from 'cozy-ui/transpiled/react/MuiCozyTheme/ListItem'
+import ListItemIcon from 'cozy-ui/transpiled/react/MuiCozyTheme/ListItemIcon'
+import ListItemText from 'cozy-ui/transpiled/react/ListItemText'
+import Radio from '@material-ui/core/Radio'
+import RadioGroup from 'cozy-ui/transpiled/react/RadioGroup'
+import Spinner from 'cozy-ui/transpiled/react/Spinner'
+import Typography from 'cozy-ui/transpiled/react/Typography'
+
+import { useI18n } from 'cozy-ui/transpiled/react/I18n'
+import { isQueryLoading, useClient, useQuery } from 'cozy-client'
+
+import {
+  ROOT_FOLDER_ID,
+  buildFoldersQuery,
+  toCozyDirectory,
+  toCozyOAuthClient,
+  updateDirectoriesExclusions
+} from 'lib/deviceConfigurationHelper'
+
+const hasQueryFailed = query => query && query.fetchStatus === 'failed'
+
+const FoldersLoading = () => (
+  <div className="u-ta-center">
+    <Spinner size="xxlarge" />
+  </div>
+)
+
+const FolderCheckbox = ({ onClick, checked, mixed, disabled, ...props }) => (
+  <IconButton size="small" onClick={onClick}>
+    <Checkbox
+      checked={checked}
+      mixed={mixed}
+      disabled={disabled}
+      disableRipple={true}
+      readOnly
+      {...props}
+    />
+  </IconButton>
+)
+
+const FolderRow = ({
+  name,
+  partialSync,
+  setPartialSync,
+  isIncluded,
+  toggleInclusion
+}) => (
+  <ListItem
+    divider
+    button
+    onClick={() => {
+      setPartialSync(true)
+      toggleInclusion()
+    }}
+  >
+    <ListItemIcon>
+      <Icon icon={FileTypeFolderIcon} width="24" height="24" />
+    </ListItemIcon>
+    <ListItemText>
+      <Typography variant="body1">{name}</Typography>
+    </ListItemText>
+    <FolderCheckbox checked={isIncluded} disabled={!partialSync} />
+  </ListItem>
+)
+
+const FoldersList = ({
+  folders,
+  partialSync,
+  excluded,
+  setPartialSync,
+  isIncluded,
+  toggleInclusion,
+  toggleAllInclusion
+}) => {
+  const { t } = useI18n()
+  const hasFolders = useMemo(() => folders.length > 0, [folders])
+  const classes = useMemo(
+    () => ({
+      root: cx('u-ph-1-half u-fz-tiny', hasFolders && 'u-pv-half'),
+      action: 'u-mv-0'
+    }),
+    [hasFolders]
+  )
+  const style = useMemo(() => ({
+    backgroundColor: 'var(--defaultBackgroundColor)'
+  }))
+
+  return (
+    <Card className="u-mt-1 u-p-0">
+      <CardHeader
+        title={
+          <Typography variant="subtitle1" color="textSecondary">
+            {t('configureDevice.my_cozy')}
+          </Typography>
+        }
+        action={
+          hasFolders && (
+            <FolderCheckbox
+              data-testid="toggle-all-inclusion"
+              checked={excluded.length !== folders.length}
+              mixed={excluded.length > 0 && excluded.length !== folders.length}
+              disabled={!partialSync}
+              onClick={() => {
+                setPartialSync(true)
```

Comment:
:thinking: This call seems unnecessary as `toggleAllInclusion()` will recompute this value

Reply:
Good catch! I forgot to remove those after I introduced `updateExcluded`.